src/components/BoxForm/BoxForm.jsx

```diff
@@ -0,0 +1,234 @@
+
+    // upload image directly to S3 bucket
+    await axios.put(uploadUrl, file, {
+      headers: {
+        'Content-Type': 'multipart/form-data',
+      },
+    });
+
+    // return box image url
+    const imageUrl = uploadUrl.split('?')[0];
+    return imageUrl;
+  };
+
+  const isValidZip = zip => {
+    const countries = [
+      'US',
+      'AT',
+      'BG',
+      'BR',
+      'CA',
+      'CZ',
+      'DK',
+      'FR',
+      'DE',
+      'IN',
+      'IT',
+      'IE',
+      'MA',
+      'NL',
+      'PL',
+      'PT',
+      'RO',
+      'RU',
+      'SG',
+      'SK',
+      'ES',
+      'SE',
+      'CH',
+      'GB',
+    ];
+    return countries.filter(country => isValidZipcode(zip, country)).length > 0;
+  };
+
+  const onChange = e => {
+    setFormData({ ...formData, [e.target.name]: e.target.value });
+  };
+
+  const submitForm = async () => {
+    if (files.length > 0) {
+      const imageUrl = await uploadBoxPhoto(files[0]);
+      setFormData({ ...formData, picture: imageUrl });
+    }
+    setSubmit(true);
+  };
+
+  const onSubmit = e => {
+    e.preventDefault();
+    setBoxNumberError(boxNumber === '' && true);
+    setZipCodeError((zipCode === '' || !isValidZip(zipCode)) && true);
+    setDateError(date === '' && true);
+    if (!(boxNumberError || dateError || zipCodeError)) {
+      submitForm();
+    }
+  };
+
+  return (
+    <form className="boxForm" onSubmit={e => onSubmit(e)}>
+      <div className="a">
+        <div className="boxInfo">
+          <FormControl isInvalid={dateError}>
+            <FormLabel htmlFor="date">Date</FormLabel>
+            <Input type="date" id="date" name="date" value={date} onChange={e => onChange(e)} />
+            {dateError && (
+              <FormErrorMessage>Invalid month, please enter a valid month</FormErrorMessage>
+            )}
+          </FormControl>
+          <FormControl isInvalid={boxNumberError}>
+            <FormLabel htmlFor="boxNumber">Box Number</FormLabel>
+            <Input
+              id="boxNumber"
+              placeholder="12345"
+              name="boxNumber"
+              value={boxNumber}
+              onChange={e => onChange(e)}
+            />
+            {boxNumberError && <FormErrorMessage>Invalid box number</FormErrorMessage>}
+          </FormControl>
+          <FormControl isInvalid={zipCodeError}>
+            <FormLabel htmlFor="zipCode">Zip Code</FormLabel>
+            <Input
+              id="zipCode"
+              placeholder="ex. 90210"
+              name="zipCode"
+              value={zipCode}
+              onChange={e => onChange(e)}
+            />
+            {zipCodeError && (
+              <FormErrorMessage>Invalid zipcode, please enter a valid zipcode</FormErrorMessage>
+            )}
+          </FormControl>
+          <FormControl>
+            <FormLabel htmlFor="location">Box Location</FormLabel>
+            <Input
+              id="location"
+              placeholder="ex. 90210"
```

Comment:
For this placeholder, we can write something different than a number again. For example, "University Park Library" or some other general location